src/scheduler.rs

```diff
@@ -245,29 +245,26 @@ impl Scheduler {
         let (smtp, smtp_handlers) = SmtpConnectionState::new();
         let (inbox, inbox_handlers) = ImapConnectionState::new();
 
-        *self = Scheduler::Running {
-            inbox,
-            mvbox,
-            sentbox,
-            smtp,
-            inbox_handle: None,
-            mvbox_handle: None,
-            sentbox_handle: None,
-            smtp_handle: None,
-        };
-
         let (inbox_start_send, inbox_start_recv) = channel(1);
-        if let Scheduler::Running { inbox_handle, .. } = self {
+        let mut inbox_handle = None;
+        let (mvbox_start_send, mvbox_start_recv) = channel(1);
+        let mut mvbox_handle = None;
+        let (sentbox_start_send, sentbox_start_recv) = channel(1);
+        let mut sentbox_handle = None;
+        let (smtp_start_send, smtp_start_recv) = channel(1);
+
+        if ctx.get_config_bool(Config::InboxWatch).await {
```

Comment:
i think, this condition also stops imap-jobs from being processed (Housekeeping, MarkseenMsgOnImap, MoveMsg, DeleteMsgOnImap etc.).

i tried a fix at https://github.com/deltachat/deltachat-core-rust/pull/1657

Reply:
you are very right, merged your fix